src/Engine/Data/Texture.hpp

```diff
@@ -56,28 +75,24 @@ struct TextureParameters {
     GLenum internalFormat { GL_RGB };
     /// Type of the components in external data
     GLenum type { GL_UNSIGNED_BYTE };
-    /// OpenGL wrap mode in the s direction
-    GLenum wrapS { GL_CLAMP_TO_EDGE };
-    /// OpenGL wrap mode in the t direction
-    GLenum wrapT { GL_CLAMP_TO_EDGE };
-    /// OpenGL wrap mode in the p direction
-    GLenum wrapP { GL_CLAMP_TO_EDGE };
-    /// OpenGL minification filter ( GL_LINEAR or GL_NEAREST or GL_XXX_MIPMAP_YYY )
-    GLenum minFilter { GL_LINEAR };
-    /// OpenGL magnification filter ( GL_LINEAR or GL_NEAREST )
-    GLenum magFilter { GL_LINEAR };
-    /// External data (ownership is left to caller, not stored after OpenGL texture creation).
-    /// Note that, for cube-map texture, this is considered as a "void*[6]" array containing the 6
-    /// faces of the cube corresponding to the targets. <br/>
-    /// texels[0] <-- GL_TEXTURE_CUBE_MAP_POSITIVE_X <br/>
-    /// texels[1] <-- GL_TEXTURE_CUBE_MAP_NEGATIVE_X <br/>
-    /// texels[2] <-- GL_TEXTURE_CUBE_MAP_POSITIVE_Y <br/>
-    /// texels[3] <-- GL_TEXTURE_CUBE_MAP_NEGATIVE_Y <br/>
-    /// texels[4] <-- GL_TEXTURE_CUBE_MAP_POSITIVE_Z <br/>
-    /// texels[5] <-- GL_TEXTURE_CUBE_MAP_NEGATIVE_Z <br/>
-    /// @todo memory allocated for this pointer might be lost at texture deletion as ownership is
-    /// unclear.
-    void* texels { nullptr };
+
+    /// texels OR cubeMap, shared ownership
+    std::shared_ptr<void> texels { nullptr };
+    std::array<std::shared_ptr<void>, 6> cubeMap {};
```

Comment:
Why not using 
```suggestion
    /// texels OR cubeMap, shared ownership
    std::variant<std::shared_ptr<void>, std::array<std::shared_ptr<void>, 6> > texels;
```
This should simplify code below (variant has operators =, ==, ...) and make the access easier (always using texels). Validity of an ImageParameters could be checked with `texels.index() == (target != GL_TEXTURE_2D)`.

Reply:
Done, I let you resolve the conversation if the proposed modification is fine for you.

Reply:
OK, some comments in my last review (07-13-2023).